gdal/frmts/netcdf/netcdfdataset.h

```diff
@@ -1038,6 +1045,8 @@ class netCDFLayer final: public OGRLayer
         int             m_nProfileVarID;
         bool            m_bProfileVarUnlimited;
         int             m_nParentIndexVarID;
+        std::shared_ptr<nccfdriver::SGeometry>       m_simpleGeometry;
```

Comment:
this variable name and the corresponding class name are a bit misleading. This is not a simple geometry, but a simple geometry reader/iterator/...

Reply:
Originally I was planning to derive new classes off of this one, but I figured that there may have been some issues with that, so yeah I can refactor that.